airbyte-integrations/connectors/source-postgres/src/main/java/io/airbyte/integrations/source/postgres/PostgresSource.java

```diff
@@ -123,25 +126,42 @@ public AirbyteCatalog discover(JsonNode config) throws Exception {
     final List<CheckedConsumer<JdbcDatabase, Exception>> checkOperations = new ArrayList<>(super.getCheckOperations(config));
 
     if (isCdc(config)) {
-      checkOperations.add(database -> database.query(connection -> {
-        final String replicationSlot = config.get("replication_slot").asText();
-        // todo: we can't use enquoteIdentifier since this isn't an identifier, it's a value. fix this to
-        // prevent sql injection
-        final String sql =
-            String.format("SELECT slot_name, plugin, database FROM pg_replication_slots WHERE slot_name = '%s' AND plugin = '%s' AND database = '%s'",
-                replicationSlot,
-                "pgoutput",
-                config.get("database").asText());
-
-        LOGGER.info("Attempting to find the named replication slot using the query: " + sql);
-
-        return connection.prepareStatement(sql);
-      }, JdbcUtils::rowToJson));
+      checkOperations.add(database -> {
+        List<JsonNode> matchingSlots = database.query(connection -> {
+          final String sql = "SELECT * FROM pg_replication_slots WHERE slot_name = ? AND plugin = ? AND database = ?";
+          PreparedStatement ps = connection.prepareStatement(sql);
+          ps.setString(1, config.get("replication_slot").asText());
+          ps.setString(2, "pgoutput");
+          ps.setString(3, config.get("database").asText());
+
+          LOGGER.info("Attempting to find the named replication slot using the query: " + ps.toString());
+
+          return ps;
+        }, JdbcUtils::rowToJson).collect(toList());
+
+        if (matchingSlots.size() != 1) {
```

Comment:
is there a possible case where this query returns >1 result? just want to make sure this error message doesn't obscure if it is possible.

Reply:
It shouldn't, but I'll log the number returned.